client/state/utils.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import tv4 from 'tv4';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+
+/**
+ * Module variables
+ */
+
+export function isValidStateWithSchema( state, schema, checkForCycles = false, banUnknownProperties = false ) {
+	const result = tv4.validateResult( state, schema, checkForCycles, banUnknownProperties );
+	if ( ! result.valid && config( 'env' ) === 'development' ) {
```

Comment:
The only problem I ran into was that it was somewhat difficult to mock out the console.warn.bind in test. The only work around I found so far was using mockery and using a require on the reducer:
```
	before( function() {
		mockery.registerMock( 'lib/warn', () => {} );
		mockery.enable();
		items = require( '../reducer' ).items;
	} );
	after( function() {
		mockery.deregisterAll();
		mockery.disable();
	} );
```